geode-apis-compatible-with-redis/src/distributedTest/java/org/apache/geode/redis/SSLDUnitTest.java

```diff
@@ -0,0 +1,301 @@
+
+    Jedis jedis;
+    try {
+      // Create the client without a keystore
+      jedis = createClient(false, false);
+    } catch (JedisConnectionException ignored) {
+      return;
+    }
+
+    // Sometimes the client is created successfully - perhaps this is platform/JDK specific
+    assertThatThrownBy(jedis::ping).satisfiesAnyOf(
+        e -> assertThat(e.getMessage()).contains("SocketException"),
+        e -> assertThat(e.getMessage()).contains("SSLException"),
+        e -> assertThat(e.getMessage()).contains("SSLHandshakeException"));
+
+    IgnoredException.removeAllExpectedExceptions();
+  }
+
+  @Test
+  public void givenMutualAuthentication_clientErrorsWithSelfSignedCert() throws Exception {
+    IgnoredException.addIgnoredException(SSLHandshakeException.class);
+    IgnoredException.addIgnoredException("SunCertPathBuilderException");
+
+    Jedis jedis;
+    try {
+      // Create the client with a self-signed certificate
+      jedis = createClient(true, true);
+    } catch (JedisConnectionException ignored) {
+      return;
+    }
+
+    // Sometimes the client is created successfully - perhaps this is platform/JDK specific
+    assertThatThrownBy(jedis::ping).satisfiesAnyOf(
+        e -> assertThat(e.getMessage()).contains("SocketException"),
+        e -> assertThat(e.getMessage()).contains("SSLException"),
+        e -> assertThat(e.getMessage()).contains("SSLHandshakeException"));
+
+    IgnoredException.removeAllExpectedExceptions();
+  }
+
+  @Test
+  public void givenSslEnabled_clientErrors_whenUsingCleartext() {
+    IgnoredException.addIgnoredException(NotSslRecordException.class);
+
+    try (Jedis jedis = new Jedis(BIND_ADDRESS, redisPort)) {
+      assertThatThrownBy(jedis::ping)
+          .isInstanceOf(JedisConnectionException.class);
+    }
+
+    IgnoredException.removeAllExpectedExceptions();
+  }
+
+  @Test
+  public void givenServerCertificateIsRotated_clientCanStillConnect() throws Exception {
+    String newServerName = "updated-server";
+
+    try (Jedis jedis = createClient(true, false)) {
+      assertThat(jedis.ping()).isEqualTo("PONG");
+    }
+
+    // create a new certificate for the server
+    CertificateMaterial serverCertificate = new CertificateBuilder()
+        .commonName(newServerName)
+        .issuedBy(ca)
+        .sanDnsName(newServerName)
+        .generate();
+
+    CertStores serverStore = CertStores.serverStore();
+    serverStore.withCertificate("server", serverCertificate);
+    serverStore.trust("ca", ca);
+
+    // Wait for one second since file timestamp granularity may only be seconds depending on the
+    // platform.
+    GeodeAwaitility.await().during(Duration.ofSeconds(1)).until(() -> true);
+    serverStore.createKeyStore(serverKeyStoreFilename, commonPassword);
+
+    // Try long enough for the file change to be detected
+    GeodeAwaitility.await().atMost(Duration.ofSeconds(PollingFileWatcher.PERIOD_SECONDS * 2))
```

Comment:
I think maybe we need to up this duration (20 seconds) to make sure this test isn't flaky in CI?

Reply:
I bumped it to 3x (30 seconds)